compare_within_models.R

```diff
@@ -0,0 +1,74 @@
+# Gregory Way 2017
+# PanCancer Classifier
+# compare_within_models.R
+#
+# Plots pancancer classifier performance compared to within cancer
+#
+# Usage: Run by assigning where the within classifier summary is and where the
+#        Pan Cancer classifier summary is
+#
+#     R --no-save --args <within_tissue_folder> <pan_cancer_folder>
+#
+# Output:
+# Bar Plots for each comparison
+
+library(ggplot2)
+source("pancancer_util.R")
+
+option_list <- list(optparse::make_option(c("-w", "--within_dir"),
+                                          type = "character",
+                                          help = "Location of within files"),
+                    optparse::make_option(c("-p", "--pancan_summary"),
+                                          type = "character",
+                                          help = "location of pancan summary"))
+
+opt_parser <- optparse::OptionParser(option_list = option_list);
+opt <- optparse::parse_args(opt_parser);
+
+within_folder <- opt$within_dir
+pan_summary_dir <- opt$pancan_summary
+pan_summary <- file.path(pan_summary_dir, "classifier_summary.txt")
+
+# Process PanCancer Classifier and summary files
+pancan_list <- parse_summary(pan_summary)
+disease_performance <- pancan_list[['Disease specific performance']]
+pancan_df <- data.frame(disease_performance[, 'disease'])
+pancan_df$Gene <- pancan_list[['Genes']]
+pancan_df$AUROC <- pancan_list[['Disease specific performance']][, 'cv']
+pancan_df$Model <- 'Pan Cancer'
+colnames(pancan_df)[1] <- 'Disease'
+
+# Process Within Cancer Results
+within_tissue_files <- list.files(within_folder,
+                                  pattern = "classifier_summary.txt",
+                                  full.names = TRUE, recursive = TRUE)
+within_tissue_data <- data.frame()
+for (file in within_tissue_files) {
+  file_summary <- parse_summary(file)
```

Comment:
It seems that line 47-52 are the same with line 33-38. Do you want to wrap them into a function?